### PR TITLE
Golang 1.6.2

### PR DESCRIPTION
**Pull request by alarcher**
Exprimental, please provide additional testing.

Only one failure in the test suite:

--- FAIL: TestGdbPython (1.08s)
    runtime-gdb_test.go:42: gdb version 7.10
    runtime-gdb_test.go:153: print mapvar failed: Python Exception <type 'exceptions.OverflowError'> long int too large to convert to int: 
        $1 = map[string]string
FAIL
FAIL    runtime 35.506s

UPDATE: Failure is only due to use of /usr/bin/python which is 32bit, changing to /usr/bin/amd64/python makes the test pass.

### CODE REVIEW

File: components/golang/Makefile [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	GOROOT_BOOTSTRAP="$(BUILD_DIR_64)/$(COMPONENT_SRC_NAME_1)"
+
+# For convenience during installation of supplementary packages.
+COMPONENT_INSTALL_ENV += PATH="$(PROTO_DIR)$(GOROOT_FINAL)/bin:$(PATH)"
+COMPONENT_INSTALL_ENV += GOROOT="$(PROTO_DIR)$(GOROOT_FINAL)"
+COMPONENT_INSTALL_ENV += GOPATH="$(SUPP_ROOT)"
+
+include ../../make-rules/prep.mk
+include ../../make-rules/justmake.mk
+include ../../make-rules/ips.mk
```
Comment (avsej) [inline components/golang/Makefile]:
```
include $(WS_MAKE_RULES)/ips.mk
```

etc

Comment (alarcher) [inline components/golang/Makefile]: And I will move to /developer.

**Comment by xen0l** [comment]
Nice, I wanted to have a look at this. I'll give this a shot today.

**Comment by xen0l** [comment]
<pre>
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/runtime/race/race_freebsd_amd64.syso' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/runtime/race/race_linux_amd64.syso' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/dwarf/testdata/cycle.elf' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/dwarf/testdata/typedef.elf4' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/compressed-64.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/gcc-amd64-openbsd-debug-with-rela.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/go-relocation-test-gcc424-x86-64.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/go-relocation-test-gcc441-x86-64.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/go-relocation-test-gcc482-aarch64.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/go-relocation-test-gcc482-ppc64le.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/go-relocation-test-gcc492-mips64.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/go-relocation-test-gcc493-mips64le.obj' in 32-bit path
INFO pkglint001.5                 Linted message: userland.action001.2  64-bit object 'usr/lib/golang/1.6/src/debug/elf/testdata/zdebug-test-gcc484-x86-64.obj' in 32-bit path
</pre>

Shouldn't we move binaries into 64-bit paths?

**Comment by alarcher** [comment]
I do not think so, the golang distribution is made that way ...
You could argue that one should put the entire directory under amd64.
This should be part of a broader discussion regarding guidelines for delivering of 64bit applications without 32bit versions.

**Comment by alarcher** [comment]
But I agree that a transform would be sexy.

File: components/developer/golang-16/Makefile [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	(cd $(BUILD_DIR_64) && \
+	/usr/gnu/bin/find src -type f \
+		-exec /usr/gnu/bin/cp --parents {} $(COMPONENT_TARGET_DIR) \;); \
+	(cd $(BUILD_DIR_64) && $(CP) -r bin $(COMPONENT_TARGET_DIR)/); \
+	(cd $(BUILD_DIR_64) && $(CP) -r pkg $(COMPONENT_TARGET_DIR)/); \
+	($(MKDIR) $(PROTOUSRSHAREMAN1DIR));
+endef
+
+# common targets
+build:		$(BUILD_64)
```
Comment (avsej) [inline components/developer/golang-16/Makefile:153]: stock syscall is almost empty, can we define `mksysnum_solaris.pl` or other way to populate `src/syscall/zsysnum_solaris_amd64.go`?

Having this empty makes some subset of tools barely portable.

The following ruby script gives the content we need:

``` ruby
puts(<<HEADER)
// +build amd64,solaris

package syscall

const (
HEADER

syscall_h = File.readlines('/usr/include/sys/syscall.h')
calls = {}
syscall_h.each do |line|
  m = line.match(/#define\s(SYS_[^\s]*)\s+(\d+)/)
  if m
    calls[m[1].upcase] = m[2]
  end
end
width = calls.keys.max_by { |x| x.length }.length + 1
calls.each do |call, num|
  printf("\t%-#{width}s = %d\n", call, num)
end
puts ")"
```

Comment (avsej) [inline components/developer/golang-16/Makefile:153]: @alarcher or add this patch https://gist.github.com/avsej/6ac03ca7baac3a3dfa444d15d4d8834e#file-0602-update-list-of-syscall-numbers-patch

Comment (alarcher) [inline components/developer/golang-16/Makefile:153]: Added.

File: components/developer/golang-16/Makefile [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+COMPONENT_LICENSE=			BSD-style, Patent Grant
+COMPONENT_LICENSE_FILE=		$(COMPONENT_NAME).license
+
+# Go has a number of supplementary tools and packages that are located in
+# separate archives or source repositories. Because Go currently forces a
+# recompile of all packages whenever the compiler changes, building and
+# delivering them within the same build component build process is the best
+# option for now.
+
+# This version is Go is not actually packaged; it is only used to bootstrap the
```
Comment (pyhalov) [inline components/developer/golang-16/Makefile]: s/is/of/

File: components/developer/golang-16/Makefile [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+COMPONENT_TARGET_DIR= \
+	$(PROTOUSRLIBDIR)/$(COMPONENT_NAME)/$(COMPONENT_MAJOR_VERSION)
+
+# We can't do any ASLR'ing as the Go 6l linker has no understanding of the
+# -z aslr=... option. So just override where needed in the .p5m files.
+ASLR_MODE = $(ASLR_DISABLE)
+
+# Make sure the source code and object files have identical timestamps so
+# the Go compiler doesn't try to rebuild them.
+PKG_MACROS += TIMESTAMP=$(shell date +"%Y%m%dT%H%M%SZ")
```
Comment (pyhalov) [inline components/developer/golang-16/Makefile]: You don't use it. userland-gate golang uses it in transforms.

**Comment by alarcher** [comment]
Upstream delivers a go.1 manpage that requires to be generated from xml to roff.
I do not think we have such tool.

**Comment by avsej** [comment]
@alarcher which manpage are you referring?

**Comment by alarcher** [comment]
@avsej https://java.net/projects/solaris-userland/sources/gate/show/components/golang/docs?rev=5906

**Comment by avsej** [comment]
but it wasn't generated from the golang upstream. Why not pull it as is?

**Comment by alarcher** [comment]
@avsej it is converted using xml2roff in the recipe but we do not have it.

**Comment by avsej** [comment]
@alarcher can't we just use the file from there? as it is pretty small

```
'\" te
.\" Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
.TH go 1 "25 Nov 2015" "SunOS 5.11" "User Commands"
.SH NAME
go \- a tool for managing Go source code

.SH SYNOPSIS


.LP
.nf
/usr/bin/go subcommand [arguments]
.fi


.LP
.nf
/usr/bin/go doc
.fi


.LP
.nf
/usr/bin/go doc <pkg>
.fi


.LP
.nf
/usr/bin/go doc <sym>[\&.<method>]
.fi


.LP
.nf
/usr/bin/go doc [<pkg>]\&.<sym>[\&.<method>]
.fi


.LP
.nf
/usr/bin/go doc <pkg> <sym>[\&.<method>]
.fi


.LP
.nf
/usr/bin/go help
.fi


.LP
.nf
/usr/bin/go help [subcommand]
.fi


.LP
.nf
/usr/bin/go help [topic]
.fi


.LP
.nf
/usr/bin/go version
.fi

.SH DESCRIPTION

.sp
.LP
\fBgo\fR is a tool for managing Go source code\&.  The \fBgo doc\fR subcommand gives you access to all of the documentation that comes with Go\&.  You can get more documentation, tutorials and community support online at <https://golang\&.org/>\&.

.SH SUB-COMMANDS

.sp
.LP
Only a few subcommands are documented here; refer to \fBgo help\fR for a complete list of subcommands and topics, and refer to \fBgo doc go\fR for complete documentation:

.sp
.ne 2
.mk
.na
\fB\fBgo help\fR \fI[subcommand|topic]\fR\fR
.ad
.br
.sp .6
.RS 4n
Without any arguments, displays a list of subcommands and available topics\&.  Use \fBgo help \fIsubcommand\fR\fR for more information about a subcommand\&.
.RE

.sp
.ne 2
.mk
.na
\fB\fBgo doc\fR \fI[pkg|sym]\fR\fR
.ad
.br
.sp .6
.RS 4n
Show documentation for package or symbol, or source in current directory if none specified\&.
.RE

.sp
.ne 2
.mk
.na
\fB\fBgo version\fR\fR
.ad
.br
.sp .6
.RS 4n
print Go version\&.
.RE
```

**Comment by alarcher** [comment]
@avsej we may, it was more a question of time than anything.

**Comment by avsej** [comment]
I've installed and built couple of projects. It seems working

**Comment by xen0l** [comment]
I also built few projects (consul, syncthing and it worked)

**Pull request merged**